test/master-playlist-controller.test.js

```diff
@@ -376,81 +376,102 @@ QUnit.test('resets everything for a fast quality change', function(assert) {
 });
 
 QUnit.test('seeks in place for fast quality switch on non-IE/Edge browsers', function(assert) {
+  let seeks = 0;
+
   this.masterPlaylistController.mediaSource.trigger('sourceopen');
   // master
   this.standardXHRResponse(this.requests.shift());
   // media
   this.standardXHRResponse(this.requests.shift());
+  // segment
+  this.standardXHRResponse(this.requests.shift());
+  this.masterPlaylistController.mediaSource.sourceBuffers[0].trigger('updateend');
 
   // media is changed
   this.masterPlaylistController.selectPlaylist = () => {
     return this.masterPlaylistController.master().playlists[0];
   };
 
-  // makes sure the resetEverything callback is queued when sourceUpdater_.remove() gets called
-  this.masterPlaylistController.mainSegmentLoader_.sourceUpdater_.processedAppend_ = true;
+  this.player.tech_.on('seeking', function() {
+    seeks++;
+  });
 
   const timeBeforeSwitch = this.player.currentTime();
 
   this.masterPlaylistController.fastQualityChange_();
-
   this.masterPlaylistController.mediaSource.sourceBuffers[0].trigger('updateend');
```

Comment:
Is this responding to an actual "append?" I think if the fastQualityChange happened, there would still have to be a segment request before an append would happen.